packages/xstate-wallet/src/chain.ts

```diff
@@ -31,43 +64,36 @@ export class ChainWatcher implements IChain {
     const response = await signer.sendTransaction(transactionRequest);
     await response.wait();
   }
-  public async getHoldings(channelId: string): Promise<string> {
+
+  public async getChainInfo(channelId: string): Promise<ChannelChainInfo> {
     const provider = getProvider();
     const contract = new ethers.Contract(
       ETH_ASSET_HOLDER_ADDRESS,
       EthAssetHolderInterface,
       provider
     );
     const amount: ethers.utils.BigNumber = await contract.holdings(channelId);
-    return amount.toHexString();
+    // TODO: Fetch other info
+    return {
+      amount,
+      finalized: false
+    };
   }
-  public on(eventType: ChainEventType, listener: ChainEventListener) {
-    if (eventType !== 'DEPOSITED') {
-      throw new Error(`No support for ${eventType}`);
-    }
+  public chainUpdatedFeed(channelId: string): Observable<ChannelChainInfo> {
     if (!this._contract) {
-      throw new Error('Chain must be initialized before being used');
-    } else {
-      const contractListener = (fromAddress, toAddress, value, event) => {
-        const chainEvent: ChainEvent = {
-          type: 'DEPOSITED',
-          channelId: event.args.destination,
-          amount: event.args.amountDeposited,
-          total: event.args.destinationHoldings
-        };
-        listener(chainEvent);
-      };
-      this._contract.on('Deposited', contractListener);
-      return () => {
-        this._contract?.removeListener('Deposited', contractListener);
-      };
+      throw new Error('Not connected to contract');
     }
-  }
-}
 
-export async function getHoldings(channelId: string): Promise<string> {
-  const provider = getProvider();
-  const contract = new ethers.Contract(ETH_ASSET_HOLDER_ADDRESS, EthAssetHolderInterface, provider);
-  const amount: ethers.utils.BigNumber = await contract.holdings(channelId);
-  return amount.toHexString();
+    const first = from(this.getChainInfo(channelId));
+
+    const updates = fromEvent(this._contract, 'DEPOSITED').pipe(
+      filter((event: {toAddress: string; amount: BigNumber}) => {
+        return event.toAddress === channelId;
+      }),
+      map(event => {
+        return {amount: event.amount, finalized: false};
+      })
+    );
+    return concat(first, updates);
```

Comment:
💡 You can also use [startWith](https://rxjs-dev.firebaseapp.com/api/operators/startWith)

```
  public chainUpdatedFeed(channelId: string): Observable<ChannelChainInfo> {
    if (!this._contract) throw new Error('Not connected to contract');

    return fromEvent(this._contract, 'DEPOSITED').pipe(
      startWith(from(this.getChainInfo(channelId))),
      filter((event: {toAddress: string; amount: BigNumber}) => {
        return event.toAddress === channelId;
      }),
      map(event => {
        return {amount: event.amount, finalized: false};
      })
    );
  }
```